ToDoList-API/ToDoList.Domain/Services/RegisterService.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using ToDoList.Domain.Interfaces;
+using ToDoList.Domain.Models;
+
+namespace ToDoList.Domain.Services
+{
+    public class RegisterService : IRegisterService
+    {
+        private readonly IUserService _userService;
+        private readonly IPasswordHasherService _passwordHasherService;
+
+        public RegisterService(IUserService userService, IPasswordHasherService passwordHasherService)
+        {
+            _userService = userService;
+            _passwordHasherService = passwordHasherService;
+        }
+
+        public async Task SignUp(SignUpDTO signUpDto)
+        {
+            var users = await _userService.GetAll();
+
+            if (signUpDto.Password != signUpDto.ConfirmPassword)
+            {
+                throw new Exception();
```

Comment:
Don't use `Exception` type
Create more specific exceptions and provide error messages